.eslintrc

```diff
@@ -8,7 +8,8 @@
         "no-alert": 0,
         "consistent-return": 0,
         "max-len": 0,
-        "prefer-rest-params": 0
+        "prefer-rest-params": 0,
+        "no-param-reassign": 0,
```

Comment:
I'd better leave the no-param-reassign turned on since it's actually a bad practice to mutate arguments values which can lead to unexpected behavior and bugs. 
In the places where can't avoid mutating of the arguments values (due to noticeable overhead in performance) we can just add eslint-disable line.

Also, if it annoys a lot we can just make this rule _warn_ instead of _error_ to keep paying attention to the places of possible bugs.

Reply:
Agree, refactored that part.